shell/src/main/java/alluxio/shell/command/CpCommand.java

```diff
@@ -554,7 +554,16 @@ private void copyFileToLocal(AlluxioURI srcPath, AlluxioURI dstPath)
     File dstFile = new File(dstPath.getPath());
     String randomSuffix =
         String.format(".%s_copyToLocal_", RandomStringUtils.randomAlphanumeric(8));
-    File tmpDst = new File(dstFile.getAbsolutePath() + randomSuffix);
+    File tmpDst;
+    File outputFile;
+    if (dstFile.isDirectory()) {
+      tmpDst = new File(PathUtils.concatPath(dstFile.getAbsolutePath(),
+              srcPath.getName() + randomSuffix));
+      outputFile = new File(PathUtils.concatPath(dstFile.getAbsolutePath(), srcPath));
```

Comment:
this needs to be the base name of `srcPath`, not the full `srcPath`. For example, if dst is "/dir" and src is "/a/b/c", we want outputFile to be "/dir/c", not "/dir/a/b/c". We can get the base path of a file by using `PathUtils.getPathComponents()` and taking the last element. It would be useful to add a `getBasePath` method to `PathUtils`, then we could use it here.